features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/processor/RecordedDataProcessorTest.kt

```diff
@@ -908,6 +932,226 @@ internal class RecordedDataProcessorTest {
             .isInstanceOf(MobileSegment.MobileRecord.MobileFullSnapshotRecord::class.java)
     }
 
+    // endregion
+    // region MetaRecords
+
+    @Test
+    fun `M send meta and focus events W process() { snapshot then touch event }`(
+        forge: Forge
+    ) {
+        // Given
+        whenever(mockRumContextDataHandler.createRumContextData())
+            .thenReturn(initialRumContextData)
+            .thenReturn(newRumContextData)
+            .thenReturn(newRumContextData)
+
+        fakeSnapshot1.forEach {
+            val fakeFlattenedSnapshot = forge.aList {
+                getForgery(MobileSegment.Wireframe::class.java)
+            }
+            whenever(mockNodeFlattener.flattenNode(it)).thenReturn(fakeFlattenedSnapshot)
+        }
+        fakeSnapshot2.forEach {
+            val fakeFlattenedSnapshot = forge.aList {
+                getForgery(MobileSegment.Wireframe::class.java)
+            }
+            whenever(mockNodeFlattener.flattenNode(it)).thenReturn(fakeFlattenedSnapshot)
+        }
+
+        val fakeTouchData = forge.aList {
+            MobileSegment.MobileRecord.MobileIncrementalSnapshotRecord(
+                timestamp = forge.aPositiveLong(),
+                data = forge.getForgery<MobileIncrementalData.PointerInteractionData>()
+            )
+        }
+
+        fakeSnapshotItem1 = createSnapshotItem(fakeSnapshot1, usedContext = initialRumContextData)
+        fakeSnapshotItem2 = createSnapshotItem(fakeSnapshot2, usedContext = newRumContextData)
+        val fakeTouchEventItem = createTouchEventItem(fakeTouchData, usedContext = newRumContextData)
+
+        testedProcessor.processScreenSnapshots(fakeSnapshotItem1)
+        testedProcessor.processScreenSnapshots(fakeSnapshotItem2)
+        testedProcessor.processTouchEventsRecords(fakeTouchEventItem)
+
+        // Then
+        val captor = argumentCaptor<EnrichedRecord>()
+        verify(mockWriter, atLeastOnce()).write(captor.capture())
+
+        val initialSnapshot = captor.allValues[0].records
+        val endSnapshot = captor.allValues[1].records
+        val newSnapshot = captor.allValues[2].records
+        val touchEvent = captor.allValues[3].records
+
+        assertThat(initialSnapshot[0] is MobileSegment.MobileRecord.MetaRecord).isTrue
+        assertThat(initialSnapshot[1] is MobileSegment.MobileRecord.FocusRecord).isTrue
+        assertThat(endSnapshot[0] is MobileSegment.MobileRecord.ViewEndRecord).isTrue
+        assertThat(newSnapshot[0] is MobileSegment.MobileRecord.MetaRecord).isTrue
+        assertThat(newSnapshot[1] is MobileSegment.MobileRecord.FocusRecord).isTrue
+        assertThat(touchEvent[0] is MobileSegment.MobileRecord.MetaRecord).isFalse
```

Comment:
I think it is better to use `assertThat(...).isInstanceOf(...)`, because in this case the message will be more explicit in case of failure. Same applies for all tests below.